### ISSUE TITLE
Modal: Support scrolling entire modal / overlay for overflowing content.

### ISSUE THREAD

**Issue by WadePeterson** [Workday/canvas-kit#790]
## 🚀 Feature Proposal
For modals with a large amount of content that will not fit on the screen, it would be nice to have an option for the overlay itself to scroll (as opposed to always requiring the consumer to make the content of the modal itself scrollable).

An additional requirement of this feature would be that the background content should not be scrollable while the modal is open. This would also likely be useful even for modals that handle scrolling within their content.

## Motivation

Currently on PEX we have several use cases for modals where the content is user defined, and can be quite large. For example, on the Home page, we have modals for announcements that are user configured and can contain media and rich text. Likewise is true for Knowledge Base articles which can be shown in modals in many places and are almost always larger than will fit on screen at any resolution.

## Example

![modal-overlay-scroll](https://user-images.githubusercontent.com/2824704/88113274-af882e00-cb6e-11ea-8dea-200d26bc68d6.gif)

**Comment by ginna-baker** [comment]
...And once the user's browser is medium or small size, even the short content on Home needs a scrollbar.

**Comment by NicholasBoll** [comment]
Related: #786, #681

**Comment by jpante** [comment]
Once #962 is complete, this will be unblock - Waiting on v5 release

**Comment by NicholasBoll** [comment]
In v5, the following is needed:

https://codesandbox.io/s/405x1

Here's an example of content scroll instead:

https://codesandbox.io/s/kwuj2

The 2 codesandboxes were created based off #905

**Comment by NicholasBoll** [comment]
Fixed in v7
